### PR TITLE
build: adding a build watch command for debugging with npm link in external apps

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** n/a

This new command is useful for debugging calcite in say the WebViewer App when using `npm link` to symlink the calcite repo.  `npm start` works for the initial build, but fails to update after changes are made, but this new watch script, which just calls `stencil build --watch` works much more consistently.

After merging this, will update https://devtopia.esri.com/WebGIS/arcgis-webviewer-app/wiki/How-to-live-debug-calcite-components-and-arcgis-components-in-arcgis-webviewer-app to use this new `npm run build:watch` command instead of `npm start`.

### CODE REVIEW

**Comment by paulcpederson** [comment]
@eriklharper when you say start "doesn't work consistently" what does that mean? What are the symptoms?

**Comment by eriklharper** [comment]
> @eriklharper when you say start "doesn't work consistently" what does that mean? What are the symptoms?

The symptom is that the changes don't reflect in the third party app until I run another `npm run build` manually.  I think its because the `dist` directory doesn't update properly when using the dev server like it does when just running the build script.

**Comment by paulcpederson** [comment]
Ok, I see so the difference is just leaving off `--dev` and `--serve`? Man it's hard to follow some of these tasks.

**Comment by eriklharper** [comment]
> Ok, I see so the difference is just leaving off `--dev` and `--serve`? Man it's hard to follow some of these tasks.

That's exactly right.  I tried getting both the calcite dev server to run at the same time, re-build on changes for the webviewer but I think the integrated dev server takes over the build and keeps the standard build from completing.  This only works if I leave off the `--dev --serve` commands.

**Comment by eriklharper** [comment]
Any objections to merging this? @Esri/calcite-components

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
   ],
   "scripts": {
     "build": "npm run util:copy-icons && stencil build && npm run util:add-custom-elements-index-mjs",
+    "build:watch": "npm run util:copy-icons && stencil build --watch && npm run util:add-custom-elements-index-mjs",
```
Comment (jcfranco) [inline package.json:20]: Are there other use cases besides the npm link usage? If not, maybe the name could reflect this?

Comment (eriklharper) [reply package.json:20]: Not that I can imagine right now.  It's only useful to me right now for rebuilding on changes for use in external apps.  I suppose it could also be used when you're just trying to serve out calcite `dist` with a generic web server running locally as well.

Comment (eriklharper) [reply package.json:20]: Kind of like a `npm start:prod` mode perhaps...

Comment (paulcpederson) [reply package.json:20]: `build:watch` is clearer to me, `start:prod` infers you're passing the `--prod` flag to the stencil compiler

Comment (jcfranco) [reply package.json:20]: Good point. I was just wondering if there was a way we could make it clearer when you would need to use that script. This could be simplified if we could bundle the icon assets differently and have the custom elements output properly output the mjs file. Then it'd just be a matter of passing flags to the stencil build script.

nothing-to-see

**Review by jcfranco — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (eriklharper) [reply package.json:20]: I like the sound of that.  Since I'm not as familiar with how those other scripts work, feel free to suggest a better way.  I'd love to better understand the purpose of each of these tasks too and why they're needed.
